packages/common/src/web3Proofs.ts

```diff
@@ -0,0 +1,596 @@
+    }
+    if (properties?.type === 'string') {
+        if (typeof statement.lower != 'string') {
+            throw new Error(
+                properties.title +
+                    ' is a string property and therefore the lower end of a range statement must be a string'
+            );
+        }
+        if (typeof statement.upper != 'string') {
+            throw new Error(
+                properties.title +
+                    ' is a string property and therefore the upper end of a range statement must be a string'
+            );
+        }
+    } else if (properties?.type === 'integer') {
+        if (typeof statement.lower != 'bigint') {
+            throw new Error(
+                properties.title +
+                    ' is a integer property and therefore the lower end of a range statement must be a bigint'
+            );
+        }
+        if (typeof statement.upper != 'bigint') {
+            throw new Error(
+                properties.title +
+                    ' is a integer property and therefore the upper end of a range statement must be a bigint'
+            );
+        }
+    }
+    if (statement.upper < statement.lower) {
+        throw new Error('Upper bound must be greater than lower bound');
+    }
+}
+
+function verifySetStatement(
+    statement: MembershipStatementV2 | NonMembershipStatementV2,
+    typeName: string,
+    properties?: PropertyDetails
+) {
+    if (statement.set === undefined) {
+        throw new Error(typeName + 'statements must contain a lower field');
+    }
+    if (statement.set.length === 0) {
+        throw new Error(typeName + ' statements may not use empty sets');
+    }
+
+    if (
+        properties?.type === 'string' &&
+        !statement.set.every((value) => typeof value == 'string')
+    ) {
+        throw new Error(
+            properties.title +
+                ' is a string property and therefore the members of a set statement must be strings'
+        );
+    }
+    if (
+        properties?.type === 'integer' &&
+        !statement.set.every((value) => typeof value == 'bigint')
+    ) {
+        throw new Error(
+            properties.title +
+                ' is a integer property and therefore the members of a set statement must be bigints'
+        );
+    }
+}
+
+function verifyAtomicStatement(
+    statement: AtomicStatementV2,
+    existingStatements: AtomicStatementV2[],
+    schema?: VerifiableCredentialSubject
+) {
+    if (statement.type === undefined) {
+        throw new Error('Statements must contain a type field');
+    }
+    if (statement.attributeTag === undefined) {
+        throw new Error('Statements must contain an attributeTag field');
+    }
+    if (
+        existingStatements.some(
+            (v) => v.attributeTag === statement.attributeTag
+        )
+    ) {
+        throw new Error('Only 1 statement is allowed for each attribute');
+    }
+
+    if (statement.attributeTag === 'id') {
```

Comment:
The `id` field was moved outside of the `attributes` portion, so I believe we allow for `id` now.